resources/images/filters.go

```diff
@@ -33,6 +35,46 @@ func (*Filters) Overlay(src ImageSource, x, y interface{}) gift.Filter {
 	}
 }
 
+// Text creates a filter that draws text with the given options.
+func (*Filters) Text(text string, options ...interface{}) gift.Filter {
+	tf := textFilter{
+		text:        text,
+		color:       "#ffffff",
+		size:        20,
+		x:           10,
+		y:           10,
+		linespacing: 2,
+	}
+
+	if len(options) > 0 {
+		opt, err := maps.ToStringMapE(options[0])
+		if err != nil {
+			panic(err)
+		}
+
+		for option, v := range opt {
+			switch option {
+			case "color":
+				tf.color = cast.ToString(v)
+			case "size":
+				tf.size = cast.ToFloat64(v)
+			case "x":
+				tf.x = cast.ToInt(v)
+			case "y":
+				tf.y = cast.ToInt(v)
+			case "linespacing":
+				tf.y = cast.ToInt(v)
+			}
+
+		}
+	}
+
+	return filter{
+		Options: newFilterOpts(text, options),
```

Comment:
@vanbroup just to be clear re the cache key vs font; the `options`  map cannot contain a pointer (unless it implements a Hashable interface, but let's not go down that route her), as that will change on every build/restart. So you need to replace the `font` key with its `.Key` value.

Reply:
> So you need to replace the font key with its .Key value.
Do you mean the `resource.Key`? How would that allow me to access the font later on?

Reply:
If you do:

```
{{ $font := resource.Get "https://example.org/font.ttf" }}
{{ $img = $img | images.Filter (images.Text "foo" (text font: $font ))}}
```

Then 

1. `$font` gets cached after first invocation.  
2. The images.Text filter will be invoked once for each property combination (which is why we want a stable font Key).


But note that if the font loading (from the Reader) in itself is an expensive operation, that is not covered by the above.

Reply:
In the last commit I updated the font value for `newFilterOpts` to `$font.Key()`, is that what you mean? 

> `{{ $img = $img | images.Filter (images.Text "foo" (text font: $font ))}}`

did you mean:

`{{ $img = $img | images.Filter (images.Text "foo" (dict "font" $font ))}}`

p.s. Feel free to push directly to this branch, as maintainer you should have the permission to do so, I don't want to keep bothering you while I'm learning the logic of the code.